packages/cozy-authentication/src/steps/ButtonLinkRegistration.jsx

```diff
@@ -1,32 +1,77 @@
-import React from 'react'
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
 import { getPlatform } from 'cozy-device-helper'
 import { Button } from 'cozy-ui/transpiled/react'
-import flag from 'cozy-flags'
+//import flag from 'cozy-flags'
 
 import { nativeLinkOpen } from '../LinkManager'
-export const ButtonLinkRegistration = ({
-  className = '',
-  label,
-  size,
-  subtle = false,
-  type = 'submit'
-}) => {
-  const url = flag('beta-onboarding')
-    ? `https://staging-manager.cozycloud.cc/cozy/create?domain=cozy.works?pk_campaign=drive-${getPlatform() ||
-        'browser'}`
-    : `https://manager.cozycloud.cc/cozy/create?pk_campaign=drive-${getPlatform() ||
-        'browser'}`
-  return (
-    <Button
-      onClick={() => {
-        nativeLinkOpen({ url })
-      }}
-      href={url}
-      label={label}
-      size={size}
-      className={className}
-      subtle={subtle}
-      type={type}
-    />
-  )
+
+import {
+  generateOAuthForUrl,
+  clearState,
+  clearSecret
+} from '../utils/onboarding'
+
+export class ButtonLinkRegistration extends Component {
+  state = {
+    url: ''
+  }
+  async generateUrl() {
+    await clearState()
+    await clearSecret()
+    const {
+      clientName,
+      redirectURI,
+      softwareID,
+      softwareVersion,
+      clientURI,
+      logoURI,
+      policyURI,
+      scope
+    } = this.props.onboarding.oauth
+    const onboardingObject = await generateOAuthForUrl({
+      clientName,
+      redirectURI,
+      softwareID,
+      softwareVersion,
+      clientURI,
+      logoURI,
+      policyURI,
+      scope
+    })
+    const url = `https://manager.cozycloud.cc/cozy/create?pk_campaign=drive-${getPlatform() ||
+      'browser'}&onboarding=${onboardingObject}`
+    this.setState({ url })
```

Comment:
It's weird to have a setState here when `generateURL` returns something.

It is easier to reason about code when functions either have side effects or are pure. A consequence is that a function should either return something or have a side effect but not both.

I do not really understand why we need to setState since `state.url` is used for the Button href but since we use onClick, it seems we do not need `href`.